### PR TITLE
Added page to view list of users and profile view. Added new user mod…

### PR DESCRIPTION
**Pull request by NikitaTitov**
…el with few custom fields

### CODE REVIEW

File: frontend/src/app/components/models/users-data.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    email: string,
+    telephone: number,
+    profile_img: string,
+    profile_url: string
+  }
+
+  interface Client extends SimpleUser {
+  }
+
+  interface Trainer extends SimpleUser {
```
[the remark this answers is not in the document]
Comment (aiharisov) [reply frontend/src/app/components/models/users-data.ts]: i'm not sure that it's a good idea to have one folder for all models

File: django_learning/settings/base.py [hunk 2]
```diff
@@ -27,13 +27,14 @@
 
 INSTALLED_APPS = [
     'rest_framework',
-    'workout_portal',
+    'workout_portal.apps.WorkoutPortalConfig',
```
Comment (introquest) [inline django_learning/settings/base.py]: beter use **workout_portal**

File: frontend/angular.json [hunk 3]
```diff
@@ -27,7 +27,8 @@
               "src/assets"
             ],
             "styles": [
-              "src/styles.scss"
+              "src/styles.scss",
+              "node_modules/bootstrap/dist/css/bootstrap.css"
```
Comment (introquest) [inline frontend/angular.json]: css from **node_modules** should include use different way

File: frontend/src/app/components/admin-page/admin-routing.module.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+import {AdminPageComponent} from "./admin-page.component";
+import {UserProfileComponent} from "./user-profile/user-profile.component";
+
+const routes: Routes = [
+  { path: 'user', component: AdminPageComponent },
+  { path: 'user/:id', component: UserProfileComponent }
```
Comment (introquest) [inline frontend/src/app/components/admin-page/admin-routing.module.ts]: Earlier this url was under control of an admin panel?

File: frontend/src/app/components/admin-page/user-profile/user-profile.component.html [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<div class="container justify-content-md-center">
+  <div class="row">
+    <label for="userName" class="col col-form-label">Name</label>
+    <div class="col">
+      <input id="userName" type="text" readonly class="form-control-plaintext" value="{{user?.username}}"/></div>
```
Comment (introquest) [inline frontend/src/app/components/admin-page/user-profile/user-profile.component.html]: why you use **?.**, user can be undefined?

File: frontend/src/app/service/admin-data.service.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  providedIn: 'root'
+})
+export class AdminDataService {
+  usersApiUrl = environment.usersAPI;
+
+  constructor(private http: HttpClient) {
+  }
+
+  getUsersList(): Observable<UserData.SimpleUser[]> {
+    return this.http.get<UserData.SimpleUser[]>(this.usersApiUrl).pipe(shareReplay(1));
```
Comment (introquest) [inline frontend/src/app/service/admin-data.service.ts]: it`s like a hint - realy you have not **UserData.SimpleUser[]**

File: frontend/src/app/service/admin-data.service.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  getUserById(id: string): Observable<UserData.SimpleUser> {
+    return this.http.get<UserData.SimpleUser>(this.usersApiUrl + id).pipe(shareReplay(1))
```
Comment (introquest) [inline frontend/src/app/service/admin-data.service.ts]: when you **shareReplay** withoup any conditions you never get repeated request

File: workout_portal/serializers/user_serializer.py [hunk 8]
```diff
@@ -1,8 +1,9 @@
...
 
+from workout_portal.models import SimpleUser
+
 
 class UserSerializer(serializers.HyperlinkedModelSerializer):
     class Meta:
-        model = User
-        fields = ('url', 'username', 'email', 'groups')
+        model = SimpleUser
+        fields = ('id', 'profile_img', 'username', 'email', 'groups', 'profile_url', 'telephone')
```
Comment (introquest) [inline workout_portal/serializers/user_serializer.py]: If you need all the fields use a special constant or do not list them.

**Pull request merged**